### PR TITLE
Add scroll to anchor on popstate (ie back/forward browser buttons)

### PR DESCRIPTION
**Pull request by RichieEscarez**
Fix for when navigating through your browser history with the back/forward buttons (to bring you to the corresponding anchor you previously clicked on).

Test in my fork: http://richieescarez.github.io/kubernetes/v1.0/

### CODE REVIEW

**Comment by thockin** [comment]
It doesn't work for me.  Going back takes me to top-of-page, though the URLbar shows the anchor.  Happy to do a screencast if you need.

**Comment by RichieEscarez** [comment]
Hmm strange. Somethings going on, its working like a charm for me in both the fork and my local build. 

Is it browser related? Im using chrome and have tried several different pages (and their anchors).

What pages/urls are you testing? Im curious if you've found one of the link thats not working right (or a new issue)? 

I know that some URLs with anchors dont work (and that there are issues open for fixing the TOC munge scripts to correctly handle characters like '?') for example this isnt working:

http://richieescarez.github.io/kubernetes/v1.0/docs/user-guide/quick-start.html#whats-next
http://kubernetes.io/v1.0/docs/user-guide/quick-start.html#whats-next

**Comment by thockin** [comment]
http://richieescarez.github.io/kubernetes/v1.0/docs/admin/node.html#node-capacity

click "Node Capacity"

scroll back to top

click "Node Status"

click the back arrow.  It takes me to TOC.  forward arrow takes me to Node Status.

**Comment by RichieEscarez** [comment]
Oh interesting. Its remembering where you last scrolled to. I tried moving/scrolling the pages and clicking back/forward a couple times and found that each time i navigated back or forth, it will bring me back to where i last scrolled to.

Ill have to look into whats happening here.

**Comment by RichieEscarez** [comment]
Ok ive solved this (and figured out how i missed this before). Apparently my local builds and even the fork seem to behave differently from the k8s.io site in that the popstate gets tracked properly and the native back/forward behaviors work when tested in my build and fork but not out on k8s.io.

Ive consolidated some code and correctly override both the popstate and normal navigation (ie clicking on something) with the same functions, see "loadandscrollpage( )" and "displaycodesample( )".

I also found and fixed the active container topic toggling. I noticed that anchors were triggering a toggle so i add a check to stop it.

File: _layouts/docwithnav.html [hunk 1]
```diff
@@ -186,8 +149,39 @@
 			}
 		    return false; // don't actually follow the link (avoid reloading page (default browser click behavior))
 	    });
+		/* Load the contents of the target page. Load the dynamic TOC otherwise scroll if there's an anchor */
```
Comment (RichieEscarez) [inline _layouts/docwithnav.html]: consolidated both popstate and click behavior into this function

File: _layouts/docwithnav.html [hunk 2]
```diff
@@ -186,8 +149,39 @@
...
+		function loadandscrollpage(targethref){
+			var navitem = findcurnavitem(),
+				offset = 0; //set default offset to the top of the page
+			$('#doccontent').load(targethref + ' #doccontent > *', function() {
+				if (navitem.hasClass('gentoc')) { //load the in-page TOC from the nav menu
+					var tocid = String(navitem.parent().children('ul').attr('id'));//cast id to string
+					$('#toclist').load( location.pathname + " #" + tocid + " li" );
+				} else if (targethref.indexOf('#') > 0) {//change offset if URL has anchor
```
Comment (RichieEscarez) [inline _layouts/docwithnav.html]: here is where we check for anchors and scroll

File: _layouts/docwithnav.html [hunk 3]
```diff
@@ -173,12 +137,11 @@
...
-					if(!isCode){//dont change/animate nav menu when dispalying a code sample
-						if(clickedon.parents().hasClass('bodywrapper')){//animate navigation menu
-							animatenav(findcurnavitem());//is an in-page link
-						} else {
-							animatenav(clickedon);//is a nav menu link
+					if(!isCode){//animate nav menu, but not when dispalying a code sample
+						if(clickedon.parents().hasClass('bodywrapper')){//if its an in-page link, first find the nav menu item
+							clickedon = findcurnavitem();
 						}
+						animatenav(clickedon, isAnchor);
```
Comment (RichieEscarez) [inline _layouts/docwithnav.html]: add the "isAnchor" attribute to stop anchors from toggling parent containers in the nav

i wont call them out but you'll also see this attribute added in two other places below

File: _layouts/docwithnav.html [hunk 4]
```diff
@@ -186,8 +149,39 @@
...
+				} else if (targethref.indexOf('#') > 0) {//change offset if URL has anchor
+					var hash = "#" + targethref.substring(targethref.indexOf("#")+1);//retrieve only the anchor from the URL
+					if ($(hash).length) {//make sure its not an empty anchor
+						offset = $(hash).offset().top - 155;
+					}
+				}
+				$('html, body').animate({scrollTop: offset}, 0);
+			});
+		}
+		/* Load the code samples in a syntax formatted code block */
```
Comment (RichieEscarez) [inline _layouts/docwithnav.html]: i also consolidated this behavior into a single function (changed nothing but moved two similar instances of this code into one function)

**Comment by RichieEscarez** [comment]
hold on. code samples are not working in the fork like they are locally

**Comment by RichieEscarez** [comment]
Hi @thockin, just pushing to the top of your inbox. 

Also, to test samples, you can cherry pick this into your local build (although i dont think its necessary since ive tested in my local build plus we know it was working out on k8s.io before i consolidated the code).

**Comment by thockin** [comment]
LGTM

**Pull request merged**